app/src/main/java/de/danoeh/antennapod/activity/BugReportActivity.java

```diff
@@ -58,12 +69,36 @@ protected void onCreate(Bundle savedInstanceState) {
                 filename.createNewFile();
                 String cmd = "logcat -d -f " + filename.getAbsolutePath();
                 Runtime.getRuntime().exec(cmd);
-                Snackbar.make(findViewById(android.R.id.content),
-                        filename.getAbsolutePath(), Snackbar.LENGTH_SHORT).show();
+
+                //share file
+                try {
+                    Intent shareIntent = new Intent(Intent.ACTION_SEND);
+                    shareIntent.setType("text/*");
+                    Uri fileUri = FileProvider.getUriForFile(this, this.getString(de.danoeh.antennapod.core.R.string.provider_authority),
+                            new File(filename.getAbsolutePath()));
+                    shareIntent.putExtra(Intent.EXTRA_STREAM,  fileUri);
+                    shareIntent.addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION);
+                    if (Build.VERSION.SDK_INT <= Build.VERSION_CODES.KITKAT) {
+                        List<ResolveInfo> resInfoList = this.getPackageManager().queryIntentActivities(shareIntent, PackageManager.MATCH_DEFAULT_ONLY);
+                        for (ResolveInfo resolveInfo : resInfoList) {
+                            String packageName = resolveInfo.activityInfo.packageName;
+                            this.grantUriPermission(packageName, fileUri, Intent.FLAG_GRANT_READ_URI_PERMISSION);
+                        }
+                    }
+                    this.startActivity(Intent.createChooser(shareIntent, this.getString(de.danoeh.antennapod.core.R.string.share_file_label)));
+
+                }
+                catch (Exception e){
+                    e.printStackTrace();
+                    Snackbar.make(findViewById(android.R.id.content),"Failed to share file", Snackbar.LENGTH_LONG).show();
```

Comment:
I think it would be better to create a new string here with a text like `No compatible apps found`.